packages/arb-token-bridge-ui/src/components/TransferPanel/TransferPanel.tsx

```diff
@@ -371,6 +390,17 @@ export function TransferPanel() {
         if (!withdrawalConfirmation) return
       }
 
+      // confirm if the user is certain about the custom destination address, especially if it matches the connected SCW address.
+      // this ensures that user funds do not end up in the destination chain’s address that matches their source-chain wallet address, which they may not control.
+      if (
+        isSmartContractWallet &&
+        isDepositMode &&
```

Comment:
i thought about it and we probably should show this dialog at both deposit and withdrawal mode, because most likely the user only controls the SCW address on the source chain, what do you think?